systemd/system/update-ssh-keys-after-ignition.service

```
@@ -0,0 +1,13 @@
+[Unit]
+Description=Run update-ssh-keys once after Ignition
+
+[Service]
+Type=oneshot
+StandardInput=file:/etc/.ignition-result.json
+ExecCondition=/usr/bin/jq -e '.userConfigProvided == true'
+ExecCondition=/usr/bin/jq -e '.provisioningBootID | gsub(\"-\"; \"\") | . == $id' --arg id %b
+ExecStart=/bin/bash -c 'ret=0; for name in $(cut -d : -f 1 /etc/passwd); do if [ -d "$(eval echo "~$name")"/.ssh/authorized_keys.d ]; then update-ssh-keys -u "$name" || ret=1; fi; done; exit $ret'
```

Up to you if you want to use this suggestion, but would this work too?

```suggestion
ExecStart=/bin/bash -c 'ret=0; for name in $(cut -d : -f 1 /etc/passwd); do if [ -d "$(getent passwd "$name" | cut -d : -f 6)"/.ssh/authorized_keys.d ]; then update-ssh-keys -u "$name" || ret=1; fi; done; exit $ret'
```

---

Maybe we can loop over home directories directly then?

---

Updated it

---

Tested by copying the unit file to a qemu vm